akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -202,3 +211,38 @@ private final class BuiltBehavior[T](messageHandlers: Array[Case[T, T]], signalH
     }
   }
 }
+
+/**
+ * The concrete behavior
+ *
+ * INTERNAL API
+ */
+@InternalApi
+private final class BuiltClassifiedBehavior[T](
+    classifiedMessageBehaviors: Map[Class[_ <: T], Array[Case[T, T]]],
+    signalHandlers: Array[Case[T, Signal]])
+    extends ExtensibleBehavior[T] {
+
+  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] =
+    classifiedMessageBehaviors.get(msg.getClass) match {
```

Comment:
This isn't correct as far as I can see. This means that the msg class must be an exact match to what is defined in the Map, but the original behavior (with the array traversal) was match on subclasses (`isAssignableFrom`)